src/pytorch_lightning/utilities/parsing.py

```diff
@@ -162,14 +162,18 @@ def get_init_args(frame: types.FrameType) -> Dict[str, Any]:
 
 
 def collect_init_args(
-    frame: types.FrameType, path_args: List[Dict[str, Any]], inside: bool = False
+    frame: types.FrameType,
+    path_args: List[Dict[str, Any]],
+    inside: bool = False,
+    classes: Tuple[Type, ...] = (object,),
```

Comment:
Initializing the tuple to contain a class looks like a code smell to me.

You could also do:

```py
classes = (,)
...

if "__class__" in local_vars and not classes or issubclass(local_vars["__class__"], classes):
    ...
```

Reply:
What about it is a code smell? 
An empty tuple expresses that there are no classes we should search for, but on the contrary, we interpret it as looking for all of them. The special casing only makes it more difficult to understand the code now.

Reply:
I guess it's not really a problem because (1) tuples are immutable and (2) it contains a class type which is a singleton anyways. But in general this pattern is discouraged because all function calls will share the same instance

```python
def foo(arg=tuple()):
    return arg

a = foo()
b = foo()
print(a is b)
```

So it's up to you. I prefer the explicit check but feel free to revert. It's got my approval both ways :smile: